### PR TITLE
feat(Live): add a new resource for callback configuration

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
1. add a new resource for callback configuration.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST='./huaweicloud/services/acceptance/live' TESTARGS='-run=TestAccRecordCallback_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/live -v -run=TestAccRecordCallback_basic -timeout 360m -parallel 4
=== RUN   TestAccRecordCallback_basic
--- PASS: TestAccRecordCallback_basic (330.29s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/live      330.416s
```

### CODE REVIEW

File: docs/resources/live_record_callback.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+subcategory: "Live"
+---
+
+# huaweicloud_live_domain
```
Comment (Lance52259) [inline docs/resources/live_record_callback.md]: s/huaweicloud_live_domain/huaweicloud_live_record_callback

File: huaweicloud/services/acceptance/live/resource_huaweicloud_live_record_callback_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testCallBack_basic(pushDomainName, "http://mycallback.com.cn/record_notify"),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "domain_name", pushDomainName),
+					resource.TestCheckResourceAttr(rName, "type.#", "1"),
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/live/resource_huaweicloud_live_record_callback_test.go]: For the TypeList parameter, we can check each element, just a tip.

Comment (Lance52259) [inline huaweicloud/services/acceptance/live/resource_huaweicloud_live_record_callback_test.go:45]: For the TypeList parameter, we can check each element, just a tips.

File: huaweicloud/services/live/resource_huaweicloud_live_record_callback.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+	resp, err := client.ListRecordCallbackConfigs(&model.ListRecordCallbackConfigsRequest{
+		PublishDomain: &publishDomain,
+		App:           &appName,
+	})
+	if err != nil {
+		return "", fmt.Errorf("error retrieving Live callback configuration: %s", err)
+	}
+
+	if resp == nil || resp.CallbackConfig == nil || len(*resp.CallbackConfig) < 1 {
+		return "", fmt.Errorf("error retrieving Live callback configuration: %s", err)
```
Comment (Lance52259) [inline huaweicloud/services/live/resource_huaweicloud_live_record_callback.go]: this error will always be `<nil>`, pls remove it and update the description.

Comment (Lance52259) [inline huaweicloud/services/live/resource_huaweicloud_live_record_callback.go:210]: the err will always be <nil>, pls remove it and update the description.

File: docs/resources/live_record_callback.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/live_record_callback.md]: fix

File: huaweicloud/services/acceptance/live/resource_huaweicloud_live_record_callback_test.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/live/resource_huaweicloud_live_record_callback_test.go]: done

File: huaweicloud/services/live/resource_huaweicloud_live_record_callback.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/live/resource_huaweicloud_live_record_callback.go]: done

File: huaweicloud/services/live/resource_huaweicloud_live_record_callback.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^http://|^https://`),
+						"The URL must start with http:// or https://."),
+					validation.StringDoesNotMatch(regexp.MustCompile(`\?{1}`), "The URL cannot contain parameters."),
+				),
+			},
+
+			"type": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/live/resource_huaweicloud_live_record_callback.go]: suggest to use `types` instead as it's a list

**Comment by ShiChangkuo** [comment]
@hmbig2 it seems that the callback authentication feature is missing?

File: huaweicloud/services/live/resource_huaweicloud_live_record_callback.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/live/resource_huaweicloud_live_record_callback.go]: done

**Pull request merged**